### ISSUE TITLE
locationTemplate option not altering request body

### ISSUE THREAD

**Issue by ferdicus** [transistorsoft/react-native-background-geolocation#267]
Hey there, thanks for this amazing work!

I'm trying to alter the default request body with the `locationTemplate` config option as described in the documentation.
However it doesn't seem to fully change the json body. 

I'm sure I must be doing something wrong here - so apologies in advance...

## Your Environment
* Plugin version: 2.8.2
* Platform: Android
* OS version: OSX 10.12.4
* Device manufacturer / model: AVD 
* React Native version (`react-native -v`): .44
* Plugin config

```js
      BackgroundGeolocation.configure({
        desiredAccuracy: 0,
        stationaryRadius: 25,
        distanceFilter: 10,
        stopTimeout: 1,
        debug: false,
        logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
        stopOnTerminate: false,
        startOnBoot: true,
        // HTTP / SQLite config
        url: 'http://10.0.2.2:3000/api/position',
        httpRootProperty: 'data',
        locationTemplate: '{ "lat":<%= latitude %>, "lng":<%= longitude %> }',
        extras: {
          "location_extra_foo": "extra location data"
        },
        batchSync: false,
        autoSync: true,
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
          Authorization: `Bearer ${token}`
        }
      }, function(state) {
        console.log("- BackgroundGeolocation is configured and ready: ", state.enabled);

        if (!state.enabled) {
          BackgroundGeolocation.start(function() {
            console.log("- Start success");
          });
        }
      });
```

## Expected Behavior
Request body that looks similar to the one described in the documentation to `locationTemplate`:

```json
POST /api/position
{
  "data": {
    "lat": 45.5192657,
    "lng": -73.6169116,
    "location_extra_foo": "extra location data"
  }
}
```

## Actual Behavior

It does seem to respect the `httpRootProperty` but ignore the `locationTemplate`.
Additionally, extras is empty.

```json
{
  "data": {
    "event": "motionchange",
    "is_moving": false,
    "uuid": "06c5c2de-27bd-4381-942f-22659afa4abe",
    "timestamp": "2017-06-16T01:47:04.000Z",
    "odometer": 0,
    "coords": {
      "latitude": 48.1122983,
      "longitude": 16.1778,
      "accuracy": 20,
      "speed": -1,
      "heading": -1,
      "altitude": 0
    },
    "activity": {
      "type": "unknown",
      "confidence": 100
    },
    "battery": {
      "is_charging": true,
      "level": 1
    },
    "extras": {}
  }
}
```

## Steps to Reproduce
1. Configure bgGeo as described above
2. Observe requests
3. Notice the difference to the documentation

## Context
Trying to send POST request to my server with modified JSON body using the `locationTemplate` option

**Comment by christocracy** [comment]
I copied your `locationTemplate` and `extras` into my app.  I see no issue here:

(btw, the plugin automatically sets the `content-type` header to `application/json`)

```
POST /locations
{
  "content-type": "application/json; charset=utf-8",
  "user-agent": "Dalvik/2.1.0 (Linux; U; Android 7.1.2; Pixel Build/NKG47L)",
  "host": "192.168.11.200:9000",
  "connection": "Keep-Alive",
  "accept-encoding": "gzip",
  "content-length": "233"
}
Authorization: undefined
{
  "location": {
    "lat": 45.5192694,
    "lng": -73.617001,
    "location_extra_foo": "extra location data"
  },
  "device": {
    "version": "7.1.2",
    "manufacturer": "Google",
    "framework": "ReactNative",
    "platform": "Android",
    "model": "Pixel",
    "uuid": "a76924c11115eb8b"
  }
}
```

**Comment by christocracy** [comment]
Are you sure the plugin is not sending cached locations recorded *before* you applied this `locationTemplate` and your server is returning `200` response?

The plugin renders the locations with the `locationTemplate` and stores the `JSON` blob in the database.  If your server isn't returning a `200`, the plugin is going to keep attempting to `POST` those locations to your server.   When your server returns an error, the plugin gives up sending any more locations.

Are you watching the logs from the plugin?

**Comment by ferdicus** [comment]
Hey Chris, thanks for the quick reply.

I resolved this issue after the pointers you left me. 

FYI my server replied with a 422 code because of the mal-formed request body.

What I did to resolve this, was to `#destroyLocations` on startUp to have a clean slate.
Additionally I also Force Stopped the App once I closed it to make sure no background task was running (I'm sure there is a way to stop background geolocations [no pun intended] via an option).

I now do receive the request body I expect on the server.

The logs were very helpful when debugging `locationTemplate` issues, howevever (I might be mistaken) it's not showing the request body when sending a POST, does it?
It only showed that a request was sent. Is there a way to in-/ de-crease verbosity?


Again, sorry for bothering you with this issue and thanks for the great job with background-geolocations.
